File: Dopamine.MiniPlayerModule/ViewModels/MicroPlayerViewModel.cs
using Dopamine.Common.Prism;
using Microsoft.Practices.Unity;
using Prism.Commands;
using Prism.Events;

namespace Dopamine.MiniPlayerModule.ViewModels
{
    public class MicroPlayerViewModel : CommonMiniPlayerViewModel
    {
        #region Variables
        private IEventAggregator eventAggregator;
        #endregion

        #region Commands
        public DelegateCommand<bool?> MicroPlayerPlaylistButtonCommand { get; set; }
        #endregion

        #region Construction
        public MicroPlayerViewModel(IUnityContainer container) : base(container)
        {

            this.eventAggregator = container.Resolve<IEventAggregator>();

            // Commands
            this.MicroPlayerPlaylistButtonCommand = new DelegateCommand<bool?>(isPlaylistButtonChecked =>
            {
                this.eventAggregator.GetEvent<MicroPlayerPlaylistButtonClicked>().Publish(isPlaylistButtonChecked.Value);
                this.IsPlaylistVisible = isPlaylistButtonChecked.Value;
            });

            ApplicationCommands.MicroPlayerPlaylistButtonCommand.RegisterCommand(this.MicroPlayerPlaylistButtonCommand);

        }
        #endregion
    }
}

File: Dopamine.MiniPlayerModule/ViewModels/MicroPlayerViewModel.vb
Imports Dopamine.Common
Imports Microsoft.Practices.Prism.Commands
Imports Dopamine.Core.Prism
Imports Dopamine.Common.Services.Search
Imports Microsoft.Practices.Prism.PubSubEvents

Namespace ViewModels
    Public Class MicroPlayerViewModel
        Inherits CommonMiniPlayerViewModel

#Region "Variables"
        Private mEventAggregator As IEventAggregator
#End Region

#Region "Commands"
        Public Property MicroPlayerPlaylistButtonCommand As DelegateCommand(Of Boolean?)
#End Region

#Region "Construction"
        Public Sub New(iEventAggregator As IEventAggregator)

            MyBase.New()

            mEventAggregator = iEventAggregator

            ' Commands
            Me.MicroPlayerPlaylistButtonCommand = New DelegateCommand(Of Boolean?)(Sub(iIsPlaylistButtonChecked)
                                                                                       mEventAggregator.GetEvent(Of MicroPlayerPlaylistButtonClicked).Publish(iIsPlaylistButtonChecked.Value)
                                                                                       Me.IsPlaylistVisible = iIsPlaylistButtonChecked.Value
                                                                                   End Sub)

            ApplicationCommands.MicroPlayerPlaylistButtonCommand.RegisterCommand(Me.MicroPlayerPlaylistButtonCommand)

        End Sub
#End Region
    End Class
End Namespace